proposals/route-blinding.md

```diff
@@ -0,0 +1,285 @@
+
+## Tips and Tricks
+
+### Recipient pays fees
+
+It may be unfair to make payers pay more fees to accomodate the recipient's wish for anonymity.
+It should instead be the recipient that pays the fees of the blinded hops (and the payer pays the
+fees to reach the introduction point).
+
+For example, if a merchant is selling an item for `N` satoshis, it should create an invoice for
+`N-f` satoshis, where `f` is the fee of the blinded part of the route.
+
+### Dummy hops
+
+The sender knows an upper bound on the distance between the recipient and `N(0)`. If the recipient
+is close to `N(0)`, this might not be ideal. In such cases, the recipient may add any number of
+dummy hops at the beginning of the blinded route by using `N(j) = N(r)`. The sender will not be
+able to distinguish those from normal blinded hops.
+
+Note that the recipient needs to fully validate each dummy hop to detect tampering.
+
+### Wallets and unannounced channels
+
+Route blinding is particularly useful for wallets that are connected to nodes via unannounced
+channels. Such wallets would use a single blinded hop, which effectively hides their `node_id`
+and `scid` from the sender. It obviously reveals to the blinded node that the next node is the
+final recipient, but a wallet that's not online all the time with a stable IP will never be able
+to hide that information from the nodes it connects to anyway (even with rendezvous).
+
+### Blinded trampoline route
+
+Route blinding can also be used with trampoline very easily. Instead of encrypting the
+`outgoing_channel_id`, we simply need to encrypt the `outgoing_node_id`.
+
+Each trampoline node can then decrypt the `node_id` of the next node and compute `E(i)` for the
+next trampoline node. That `E(i)` can then be sent in the outer onion payload instead of using the
+message's TLV extensions, which is even cleaner.
+
+## FAQ
+
+### Why not use rendezvous
+
+While rendezvous is more private, it's also less flexible: it doesn't support reusing the partial
+onion nor retrying with updated fees on intermediate node failure. Route blinding has different
+trade-offs, which makes it useful for slightly different use-cases than rendezvous.
+
+Route blinding lets senders choose the amounts and cltv sent through each blinded channel: it makes
+payment success more likely, but introduces a probing surface for attackers.
+
+### Why not use HORNET
+
+HORNET requires a slow session setup before it can provide useful speedups. In cases where you
+expect to send a single message per session (which is the case for most payments), HORNET actually
+performs worse than Sphinx in latency, bandwidth and privacy.
+
+## Open Questions
+
+* Should we include feature bits in `encrypted_recipient_data`? It's yet another probing vector so
+  we'd need to "sanitize" them to avoid reducing the node's anonymity set...
+* Should we add the `payment_hash` to the additional data authenticated by `encrypted_recipient_data`?
```

Comment:
If we commit to the `payment_hash` (can also be implicit in the `ss(i)` derivation fwiw) we could sort of limit the free probing an intermediate node can do. Without it a node can just probe using a provided blinded route without actually spending any funds.

It is a bit limiting for sure, but makes probing non-free.

Reply:
Agreed, I think this is something that we should do!

Reply:
Done in https://github.com/lightningnetwork/lightning-rfc/pull/765/commits/0468ed01e003e1e1ed7e49c0802b5f4c10519cba